searx/engines/discourse.py

```diff
@@ -0,0 +1,95 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""Discourse Forums
+
+"""
+
+from urllib.parse import urlencode
+import html
+
+from dateutil import parser
+
+from searx.exceptions import SearxEngineException
+
+about = {
+    "website": "https://discourse.org/",
+    "wikidata_id": "Q15054354",
+    "official_api_documentation": "https://docs.discourse.org/",
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": "JSON",
+}
+
+base_url = None
+search_endpoint = '/search.json?'
+
+# ['latest', 'likes', 'views', 'latest_topic']
+api_order = 'likes'
+paging = True
+
+
+def request(query, params):
+    """https://docs.discourse.org/#tag/Search/operation/search"""
+    if len(query) <= 2:
+        raise SearxEngineException(
+            'For engines using discourse the length of the search query must be at least 3 chars!'
+        )
+
+    args = urlencode(
+        {
+            'q': query + f' order:{api_order}',
+            'page': params['pageno'],
+        }
+    )
+
+    params['url'] = base_url + search_endpoint + args
+    params['headers'].update(
+        {
+            'Accept': 'application/json, text/javascript, */*; q=0.01',
+            'X-Requested-With': 'XMLHttpRequest',
+        }
+    )
```

Comment:
```suggestion
    params['headers'] = {
        'Accept': 'application/json, text/javascript, */*; q=0.01',
        'X-Requested-With': 'XMLHttpRequest',
    )
```
params['headers'] should be empty by default, shouldn't it?